### DISCUSSION TITLE
Attempting to bind values to reserved name should trigger a warning

### DISCUSSION THREAD

**Discussion by hbina** [slint-ui/slint#738] [Ideas]
I have wasted a disgusting amount of time only to realize that `row` is a reserved keyword. I basically had something like this compiled without warning, which does not even make any sense because `row` here is shadowed by the reserved name anyways.

```60
for row in bson.inner : HorizontalBox {
    key := Text {
        text: row.key;
    }
    value := Text {
        text: row.value;
    }
}
```

**Comment by ogoffart** [comment]
True. We should probably error out if re-using reserved names.
I created issue #739 to track that.
Thanks for the report.
